### DISCUSSION TITLE
Per-second data is now back in Vuegraf

### DISCUSSION THREAD

**Discussion by jertel** [jertel/vuegraf#57] [General]
There have been several changes over the past couple of weeks related to data polling of the Emporia API, initiated by the Emporia team due to high loads on their servers from non-Emporia app users (e.g. projects based on the PyEmVue project). Full discussion can be found here: https://github.com/magico13/PyEmVue/issues/19.

The summary of this discussion, as it relates to this project is this: Vuegraf will now poll for each channel's total energy usage over the past minute, once per minute. Additionally, once per hour, Vuegraf will request each channel's energy usage for each second over the past 60 minutes. This effectively means Vuegraf provides the same data as it had been providing, with the only difference being that the per-second data will lag by 1 hour, while per-minute data will be near real-time.

Additionally, per-second data will be tagged with `detailed=True`, while per-minute data will be tagged with `detailed=False`. This will allow Influx queries to distinguish between the two data rates, if there's a desire to do this. In general that shouldn't be necessary unless calculating usage totals (sums) rather than usage averages.

**Comment by Cjkeenan** [comment]
What is your opinion on modifying the detailed interval from 3600 to another lower value like 900 (15 minutes) or even 60 so that every other minute the previous 120 seconds of data is pulled? Do you think this would be bad for the Emporia servers? Honestly, I wish they would allow a local export on the network over HTTP or something taking all of the load from these 3rd party projects from their servers.

**Reply by jertel** [reply]
From what Emporia told us it does sound like that frequent of a polling for second data does have a bad impact on their servers. I don't personally see a need for reducing the interval to that small of a value. What is your use case?

**Reply by Cjkeenan** [reply]
I don't have a direct use case since for real time data I can just use the app. I did notice, however, if there was any issue in the chain, such as the host device restarting/going down or the internet going down for instance, with detailed data being pulled every hour, there is a good chance that there will be a pretty sizable block of minute only data in the database. If you reduce that detailed interval of course, that impact is minimized.

**Comment by chase1124** [comment]
Are many users able to meaningfully use backfilled per second usage metrics? I think with a 1 hour delay this level of granularity is not that helpful -- considering the limitation I would probably use their actual app if I need per second telemetry. What I'm wondering is if it wouldn't be possible to disable this via configuration file, probably disabled by default -- it creates a lot of data most users won't need (I'm guessing) and more possibility for error when summing graphs if users (like me) didn't read the docs closely enough the first time.

**Reply by jertel** [reply]
Good suggestion. Let's see if anyone else has an opinion on it. If not, I can make that change in a future release.
